### ISSUE TITLE
exclude deprecated releases from `cabal outdated` reporting

### ISSUE THREAD

**Issue by hvr** [haskell/cabal#5603]
A recent annoyance w/ `cabal outdated` is that it when you have something like

```
build-depends: binary >= 0.7.1  && < 0.9
```

`cabal outdated` would complain about

```
Outdated dependencies:
binary >=0.7.1 && <0.9 (latest: 0.10.0.0)
```

even though `binary-0.8.6.0` is the latest valid release, and binary-0.9.0.0 and binary-0.10.0.0 have been marked as deprecated (or rather "unpreferred") in the package index.

/cc @23Skidoo

**Comment by RyanGlScott** [comment]
See also #4325.

**Comment by fendor** [comment]
Closed by #7296
